engine/execution/state/delta/view.go

```diff
@@ -119,32 +117,37 @@ func (v *View) RegisterUpdates() ([]flow.RegisterID, []flow.RegisterValue) {
 // This function will return an error if it fails to read from the underlying
 // data source for this view.
 func (v *View) Get(owner, controller, key string) (flow.RegisterValue, error) {
+	registerID := flow.NewRegisterID(owner, controller, key)
+
 	value, exists := v.delta.Get(owner, controller, key)
 	if exists {
-		// every time we read a value (order preserving) we update spock
-		var err error
-		if value != nil {
-			err = v.updateSpock(value)
+		// every time we read a value (order preserving) we update the secret
+		// with the registerID only (value is not required)
+		err := v.updateSpockSecret(registerID.Bytes())
+		if err != nil {
+			return nil, fmt.Errorf("get register failed: %w", err)
 		}
-		return value, err
+		return value, nil
 	}
 
 	value, err := v.readFunc(owner, controller, key)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("get register failed: %w", err)
 	}
 
-	registerID := toRegisterID(owner, controller, key)
-
 	// capture register touch
 	v.regTouchSet[registerID.String()] = registerID
 
 	// increase reads
 	v.readsCount++
 
-	// every time we read a value (order preserving) we update spock
-	err = v.updateSpock(value)
-	return value, err
+	// every time we read a value (order preserving) we update the secret
+	// with the registerID only (value is not required)
+	err = v.updateSpockSecret(registerID.Bytes())
+	if err != nil {
+		return nil, fmt.Errorf("get register failed: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("set register failed: %w", err)
```

Reply:
I think the function is still `get`, or did I miss something? :)

Reply:
I think all operations here are register reads. hence the wording `get`